ompi/mca/coll/han/coll_han_alltoallv.c

```diff
@@ -78,11 +78,74 @@ struct gathered_data {
     size_t rtype_serialized_length;
     size_t sbuf_length;
     size_t rbuf_length;
+    ssize_t sbuf_lb;
+    ssize_t rbuf_lb;
     void *sbuf;
     void *rbuf;
     void *serialization_buffer;
 };
 
+/**
```

Comment:
yes, this sucks and all collectives have exactly the same issue. That's why we already have a function for it `opal_datatype_span`.

Reply:
Using it in new revision.